pathfinder/token_network.py

```diff
@@ -146,25 +148,29 @@ def update_balance(
         )
 
     def update_fee(
-        self,
-        channel_id: ChannelId,
-        new_fee: float,
-        signature
+            self,
+            channel_id: ChannelId,
+            new_fee: bytes,
+            signature: bytes
     ):
-        # FIXME: Signature on fee update not checked !!
-        # msg = new_fee
-        # signer = from_signature_and_message(signature, msg)
-        # <-- this verifies the signature of the fee-update!
-        signer = signature
+        msg = new_fee
+        signer = public_key_to_address(
+            PublicKey.from_signature_and_message(
+                signature,
+                msg,
+                hasher=sha256)
+        )
         participant1, participant2 = self.token_network_contract.get_channel_participants(
             channel_id
         )
         if is_same_address(participant1, signer):
             sender = participant1
             receiver = participant2
+            new_fee = float(new_fee)
```

Comment:
Shouldn't these be `int`s? We should probably define if we handle values in the smallest denomination or relative to a "full" token itself . @devium